src/cli/get.rs

```diff
@@ -0,0 +1,178 @@
+use std::{
+    env::{current_dir, temp_dir},
+    fmt::Display,
+    fs::{copy, exists, remove_file},
+    path::Path,
+    process,
+};
+use uuid::Uuid;
+
+use crate::cli::common;
+use clap::Parser;
+use regex::Regex;
+
+#[derive(Parser, Debug, Default)]
+#[command(arg_required_else_help = true)]
+pub struct Args {
+    /// URL or <github org>/<repo name> of the environment to grab
+    env: String,
+}
+
+#[derive(Debug, Default)]
+pub struct RemoteRepo {
+    org: String,
+    repo: String,
+    domain: Option<String>,
+    protocol: Option<String>,
+}
+
+impl RemoteRepo {
+    /// Render the repository as a git url
+    fn as_url(&self) -> String {
+        format!(
+            "{}{}/{}/{}",
+            self.protocol.clone().unwrap_or("https://".into()),
+            self.domain.clone().unwrap_or("github.com".into()),
+            self.org,
+            self.repo
+        )
+    }
+
+    /// Render the repository as an ssh URL
+    fn as_ssh_url(&self) -> String {
+        format!(
+            "git@{}:{}/{}.git",
+            self.domain.clone().unwrap_or("github.com".into()),
+            self.org,
+            self.repo
+        )
+    }
+}
+
+impl Display for RemoteRepo {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        write!(f, "{}", self.as_url())
+    }
+}
+
+/// Clone the given environment URL
+///
+/// * `env`: Remote URL for an environment. If only <org>/<repo> is passed, the repository is
+///   assumed to live on github.
+fn parse_repo_arg(env: &str) -> Result<RemoteRepo, String> {
+    let re = Regex::new(
+        r"((?<protocol>(git\+)?https?://)?(?<domain>github\.com)/)?(?<org>\w+)/(?<repo>\w+)",
+    )
+    .map_err(|_| "Invalid regex for processing git url.")?;
+
+    let captures = re
+        .captures(env)
+        .ok_or("Unrecognized format for repo name or URL: {env}.")?;
+
+    Ok(RemoteRepo {
+        protocol: captures
+            .name("protocol")
+            .map(|name| name.as_str().to_string()),
+        domain: captures
+            .name("domain")
+            .map(|name| name.as_str().to_string()),
+        org: captures
+            .name("org")
+            .ok_or("No org name found in {env}")?
+            .as_str()
+            .to_string(),
+        repo: captures
+            .name("repo")
+            .ok_or("No repo name found in {env}")?
+            .as_str()
+            .to_string(),
+    })
+}
+
+pub fn execute(args: Args) {
+    let cwd = match current_dir() {
+        Ok(dir) => dir,
+        Err(_) => {
+            eprintln!("Could not get the current directory.");
+            return;
+        }
+    };
+    let target_toml = cwd.join("pixi.toml");
```

Comment:
I'm not sure about this target directory being the cwd. Currently, this project is setup to put all the environment lockspecs in a global araki dir that serves as a registry of available environments. 

Tbh, I think how this all ought to work is a bit unclear to me. So I don't think this is a blocker, given that we are just trying stuff out here. But, I do think this will break some assumptions made in stuff like the `araki envs list` command.

Reply:
Thinking about this a little more, this should probably clone the repository to `~/.araki/envs/`, and then hardlink `pixi.toml` and `pixi.lock` from there to the project repository. Hmm, let me take a second crack at this.

Reply:
Okay, this has been reworked and cloning the repo now puts it in `~/.araki/envs/`. About your cwd comment :point_up: I think we can allow `get` to accept a `path` arg later on if that's what we want to do, nothing here precludes that option.